text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
As I [mentioned above](https://github.com/rust-lang/rfcs/pull/578#discussion_r23659153) the implementation does not actually have a static lifetime (to returned a borrowed Cow) on all platforms. Additionally, on platforms like OSX there is no safeguard against other libraries in the process perhaps modifying the arguments.

If we can work out an implementation to actually have a totally safe static lifetime then we have a small window where we can change the iterator to yield `CowOsString<'static>`, but it's somewhat questionable whether the work will be worth it in the long run.